tests/components/icloud/test_config_flow.py

```diff
@@ -59,6 +59,20 @@ def mock_controller_service_with_cookie():
         yield service_mock
 
 
+@pytest.fixture(name="service_with_cookie_no_device")
+def mock_controller_service_with_cookie_no_device():
+    """Mock a successful service while already authenticate, but without device."""
+    with patch(
+        "homeassistant.components.icloud.config_flow.PyiCloudService"
+    ) as service_mock:
+        service_mock.return_value.requires_2sa = False
+        service_mock.return_value.trusted_devices = TRUSTED_DEVICES
+        service_mock.return_value.send_verification_code = Mock(return_value=True)
+        service_mock.return_value.validate_verification_code = Mock(return_value=True)
+        service_mock.return_value.devices = Mock(side_effect=KeyError("findme"))
```

Comment:
As usual, the mock is not working here.
@MartinHjelmare 
How should I do to fix this, please ?

Library code : 
https://github.com/picklepete/pyicloud/blob/9bb6d750662ce24c8febc94807ddbdcdf3cadaa2/pyicloud/base.py#L291-L299

Reply:
Try using a PropertyMock instead.

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.PropertyMock

Reply:
Still not working, returning `<PropertyMock name='PyiCloudService().devices' id='4536752976'>`

Reply:
`PropertyMock(return_value=KeyError("findme"))` leads to the same result

Reply:
`required_2sa` and `trusted_devices` are also properties but `PropertyMock` wasn't the solution.

I have no clue on this.

Reply:
@MartinHjelmare 
Also tested on with a patch in the test and a `PyiCloudNoDevicesException` side_effect.

Reply:
Any help on this ?

Reply:
fixed